thunderdomebot.py

```diff
@@ -3,105 +3,174 @@
+async def get_top_messages(ctx, emoji: str = None, number: int = 5):
+    if ctx.guild is None:
+        await ctx.send("This command can only be used within a server!")
+        return
+
+    if number < 1 or number > 10:
+        await ctx.send("Number of messages must be between **1** and **10**")
+        return
+
+    if emoji is None:
+        sql_emoji_command = ""
+    else:
+        sql_emoji_command = "WHERE emoji = "+sql_string(emoji)+" "
+    cursor.execute("SELECT message_id, MAX(author_id), SUM(count) as score FROM messages {}GROUP BY message_id ORDER BY score DESC LIMIT {}".format(sql_emoji_command, number))
+    rows = cursor.fetchall()
+    
+    if len(rows) == 0:
+        await ctx.send("No messages found with any {}".format(str(emoji)))
+        return
+
+    title = "Top {} by {}".format(str(number)+" messages" if number > 1 else "message", str(emoji) if emoji is not None else "all")
+    description = ""
+    listnum = 0
+
+    for row_elements in rows:
+        print("Fetching message from {} with ID = {}".format(ctx.guild.get_member(row_elements[1]).name, row_elements[0]))
+        listnum += 1
+        for channel in ctx.guild.text_channels: # Try to get the message from each channel
+            found = False
+            try:
+                message = await channel.fetch_message(row_elements[0])
+                emoji_text = str(emoji)+" " if emoji is not None else " "
+                description += "{0}. {1.author.name} with {2}x{3} ([link]({4}))\n".format(listnum, message, row_elements[2], emoji_text, message.jump_url.strip("<>"))
+                if number <= 3:
+                    message_text = message.content
+                    if len(message_text) > 0:
+                        text_preview = (message_text[:247]+"...") if len(message_text) > 250 else message_text
+                        description += "> "+text_preview+"\n"
+                    description += "\n"
+                found = True
+                break
+            except discord.NotFound:
+                pass
+        
+        if not found:
+            description += "{}. [Message deleted/not found]\n".format(listnum)  # Print this if the message was not found in any channel
+    
+    embed = discord.Embed(title=title, description=description)
+    await ctx.send(embed=embed)
+
 
+async def read_message_history(guild, num_days = None):
+    '''Parses a server's message history, optionally stopping after num_days of messages'''
+
+    if num_days is not None:
+        cutoff_time = datetime.utcnow() - timedelta(days=num_days)
+    else:
+        cutoff_time = datetime.min()    # Read all history
 
-async def read_message_history(guild):
     for channel in guild.text_channels:
         messages_parsed = 0
         if channel.permissions_for(guild.me).read_messages:
-            async for message in channel.history(limit=100):
+            async for message in channel.history(limit=200):
```

Comment:
this means `cutoff_time` doesn't get used right?

Reply:
Yeah, I disabled it. We need to decide exactly how that logic should work. I think the bot will need to record how long it's been since the db was updated.